upb/json/printer.c

```diff
@@ -47,12 +47,19 @@ void freestrpc(void *ptr) {
 strpc *newstrpc(upb_handlers *h, const upb_fielddef *f) {
   /* TODO(haberman): handle malloc failure. */
   strpc *ret = malloc(sizeof(*ret));
-  size_t len;
-  ret->len = upb_fielddef_getjsonname(f, NULL, 0);
-  ret->ptr = malloc(ret->len);
-  len = upb_fielddef_getjsonname(f, ret->ptr, ret->len);
-  UPB_ASSERT_VAR(len, len == ret->len);
-  ret->len--;  /* NULL */
+  if (getenv("UPB_JSON_WRITE_LEGACY_FIELD_NAMES")) {
```

Comment:
Is this a performance concern? You could cache somewhere instead of looking up the env variable repeatedly